app/views/edit/build-config.html

```diff
@@ -18,49 +18,62 @@
-                              <div>
-                                <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
-                                <input class="form-control"
-                                  id="sourceUrl"
-                                  name="sourceUrl"
-                                  ng-model="updatedBuildConfig.spec.source.git.uri"
-                                  type="text"
-                                  autocorrect="off"
-                                  autocapitalize="off"
-                                  spellcheck="false"
-                                  required>
-                              </div>
-                              <div>
-                                <span class="text-warning" ng-if="form.sourceUrl.$dirty && !sourceURLPattern.test(updatedBuildConfig.spec.source.git.uri)">Git repository should be a URL.</span>
+                <div class="row">
+                  <div class="col-lg-12">
+                    <div ng-if="buildConfig.spec.source.type !== 'None'" class="section">
+                      <h3>Source Configuration</h3>
+                      <dl class="dl-horizontal left">
+                        <div ng-if="sources.git">
+                          <div class="row">
+                            <div ng-class="{
+                              'col-lg-8': advancedOptions,
+                              'col-lg-12': !advancedOptions}">
+                              <div class="form-group">
+                                <label for="sourceUrl" class="required">Git Repository URL</label>
+                                <div>
+                                  <!-- Unfortunately, we can't set type="url" because some valid values don't pass browser validation. -->
+                                  <input class="form-control"
+                                    id="sourceUrl"
+                                    name="sourceUrl"
+                                    ng-model="updatedBuildConfig.spec.source.git.uri"
+                                    type="text"
+                                    autocorrect="off"
+                                    autocapitalize="off"
+                                    spellcheck="false"
+                                    required>
+                                </div>
+                                <div>
+                                  <span class="text-warning" ng-if="form.sourceUrl.$dirty && !sourceURLPattern.test(updatedBuildConfig.spec.source.git.uri)">Git repository should be a URL.</span>
+                                </div>
+                                <div class="help-block">
+                                  Location of source code to use as an input for the build.
+                                  <span ng-if="!advancedOptions">In case your your Git repository is secured, switch to the 'advanced options' view to set up your credentials.</span>
```

Comment:
```html
If your Git repository is private, view the <a href="" ng-click="advancedOptions = true">advanced options</a> to set up authentication.
```

You probably need to add a dot into the advancedOptions scope var path since this is inside an `ng-if` block. `view.advancedOptions` or similar